code/modules/clothing/glasses/glasses.dm

```diff
@@ -370,6 +370,8 @@ BLIND     // can't see anything
 	desc = "A stylish pair of tactical goggles that protect the eyes from aerosolized chemicals, debris and bright flashes."
 	var/brand_name
 	icon = 'icons/clothing/eyes/goon_goggles.dmi'
+	icon_state = "military_goggles"
+	item_state = "military_goggles"
```

Comment:
Remember to check for inhands when changing item_state.

If none are present, use a generic parent one.

Reply:
None of the goggles changed here seem to have inhands and the closest thing to a generic parent variant is in safety.dmi, rather than goon_goggles.dmi. Would it be proper to just omit an item_state for them, or is there a good way to pull the generic inhands from the other file to use here?

Reply:
I see. That's fine then. I'd suggest migrating goon_goggles.dmi entirely inside of safety.dmi since they are a subset of regular safety goggles and the sprites are the exact same anyway, so you can get rid of the icon variable as well.

After digging in the files for a while, I now see the goggles are contained sprites. There isn't an easy way to pull the generic inhands, but you can simply copy and paste the generic version and rename the icons in the dmi to match. The icons aren't your responsibility, so you don't have to worry about changing the color. This is up to you, but it's just good practice to do so.

That aside, since they *are* contained sprites, you do not need to change the item_state. However, this still applies to non-contained sprite objects, so be sure to check.

Reply:
I've migrated all the icons in goon_goggles.dmi over to safety.dmi, all the items inherit the icon from there now. I've also found a way to overwrite the item_state for the right and left hand slots to use the generic goggle inhands, now that they're all in the same file, which seems to work fine ingame. Does that all look good to you?

Reply:
The goggles also have a funny initialisation that defines the item state off the sprite state, so I've removed the item_state definitions.